src/benchmarks/micro/corefx/System.Linq/LinqTestData.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Linq.Tests
+{
+    public class LinqTestData
+    {
+        // this field is a const (not instance field) to avoid creating closures in tested LINQ
+        internal const int Size = 100;
+
+        private static readonly int[] _arrayOf100Integers = Enumerable.Range(0, Size).ToArray();
+
+        internal static readonly LinqTestData Array = new LinqTestData(_arrayOf100Integers);
+        internal static readonly LinqTestData List = new LinqTestData(new List<int>(_arrayOf100Integers));
+        internal static readonly LinqTestData Range = new LinqTestData(Enumerable.Range(0, Size));
+        internal static readonly LinqTestData IEnumerable = new LinqTestData(new IEnumerableWrapper<int>(_arrayOf100Integers));
```

Comment:
Why not `.AsEnumerable()`?

It's annoying that we differentiate `IList<T>` and List`<T>` in the BCL code, so we need both representations. (And that there is no `.ToIList()`

Reply:
The goal of the `IEnumerableWrapper` is to hide the information about underlying collection type and implement `IEnumerable<T>` without implementing `List`, `IList`, `ICollection` etc

The `AsEnumerable` just returns the argument so the rest of the LINQ code could still check the type and run the code path for specific type implementation (array in this case)

https://github.com/dotnet/corefx/blob/143df51926f2ad397fef9c9ca7ede88e2721e801/src/System.Linq/src/System/Linq/Enumerable.cs#L11

```cs
List<T> list = new List<T>();
IEnumerable<T> enumerable = list.AsEnumerable();
Assert.True(enumerable is List<T>);
```